File: src/main/java/no/novari/flyt/history/model/action/InstanceStatusTransferredOverrideAction.java
package no.novari.flyt.history.model.action;

import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.NotNull;
import lombok.Builder;
import lombok.Getter;
import lombok.Setter;
import lombok.extern.jackson.Jacksonized;
import no.novari.flyt.history.model.SourceApplicationAggregateInstanceId;

@Jacksonized
@Builder
@Getter
@Setter
public class InstanceStatusTransferredOverrideAction implements SourceApplicationAggregateInstanceId {
    @NotNull
    private Long sourceApplicationId;
    @NotEmpty
    private String sourceApplicationInstanceId;
    @NotEmpty
    private String sourceApplicationIntegrationId;
}

File: src/main/kotlin/no/novari/flyt/history/model/action/InstanceStatusTransferredOverrideAction.kt
package no.novari.flyt.history.model.action

import jakarta.validation.constraints.NotEmpty
import jakarta.validation.constraints.Positive
import no.novari.flyt.history.model.SourceApplicationAggregateInstanceId

data class InstanceStatusTransferredOverrideAction(
    @field:Positive
    override val sourceApplicationId: Long = 0,
    @field:NotEmpty
    override val sourceApplicationInstanceId: String = "",
    @field:NotEmpty
    override val sourceApplicationIntegrationId: String = "",
) : SourceApplicationAggregateInstanceId
